debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java

```diff
@@ -597,6 +603,34 @@ private long asEpochMicros(String timestamp) {
 
         String expectedTzLarge = "+21016-11-04T06:51:30.123456Z";
 
+        // The assertion for minimimum timestamps is problematic as it seems that Java and PostgreSQL handles conversion from large negative date
+        // to microseconds in different way
+        // written to database: 4713-12-31T23:59:59.999999 BC
+        // arrived from decoding plugin: -210831897600000001
+        // value from decoding plugin converted to Instant: -4712-12-31T23:59:59.999999Z - 1 year difference
+        // JDBC driver delivers Timestamp B.C.E. 4713-12-31T23:59:59.000+0100 that internally contains -210835184391000001 and translates to Instant -4712-11-23T22:59:59.999999Z
+        // The result is that JDBC driver and logical decoding plugin provides different values which moreover
+        // do not map to Java conversions
+        // It seems that JDBC driver always uses Gregorian calendar while Java Time API uses Julian
+        // Also it seems that further distortions could be introduced by timezone used so instead of matching
+        // against exact value the requested date should be smaller than -4712-11-01
+        final SchemaAndValueField.Condition largeNegativeTimestamp = (String fieldName, Object expectedValue, Object actualValue) -> {
```

Comment:
Lots of good explanation, but one key part is missing: what's the approach for dealing with this? I.e. does the assertion have some leniency built in?

Reply:
@gunnarmorling Already rebased. The leniency is described at exactly tha last too lines of the comment

Reply:
Thx!